source/MainPanel.cpp

```diff
@@ -50,6 +50,33 @@ using namespace std;
 
 
 
+namespace {
+	// Comparator class for comparing two strings, assumed to represent outfit
+	// category names, by the order of those categories in the UI.
+	class CategoryNameCompare {
```

Comment:
Sorry, what's a "standalone class"? The definition "a class that's not part of an inheritance hierarchy" doesn't seem to fit here. Do you mean I should put the class definition into Outfit.cpp or CategoryNameCompare.cpp? Or do you just mean the class shouldn't rely on any state except its own members?

As to tests, can do. By the way, tests don't build on my system without a tweak. Running `scons test` fails for me with linker errors for pthread, libjpeg, and libpng symbols. I noticed in the output of `scons` and `scons test` that `endless-sky` is linked with `-luuid -lSDL2 -lpng -ljpeg -lopenal -lpthread -lGL -lGLEW -lmad` while `tests/endless-sky-tests` is linked with `-luuid` only. If I manually run the link command printed by `scons test` with all those -l flags added, it succeeds, and a subsequent `tests/endless-sky-tests -i --warn NoAssertions --order rand --rng-seed 'time'` succeeds as well (`All tests passed (983 assertions in 36 test cases)`). That's with pre- 759b3d029ff08b1eb295455168b1ac38cea0b59e master.

Reply:
By standalone I mean in its own translation unit so that it can be used by other TUs by means of a simple include.

(The tests should build fine with no tweaks unless you have disabled LTO optimization for the release build. The third party libs are deliberately not linked when building the tests, and as such you can not yet test any class method that could result in needing a GameData call.)

Reply:
Done. It actually took me more time to do this refactoring than it took me to write the feature in the first place :(

(I haven't disabled LTO. I'm not sure what's going wrong in my build, but anyway, I rewrote the comparator with std::find. Hope now it's simple enough that it doesn't need tests.)

Reply:
Tests aren't _just_ for complex things, they're documentation and also "insurance" for implementation refactoring.

What compiler and linker versions do you have?

Reply:
Added tests in e62d63d2927c25a7eea427a0a4f8797d9d3e5951 .  They seem to run and pass here when I manually link with the third party libraries. About my setup I'll answer on discord as it's off topic :)

Reply:
Added tests for the other new class in cff1eb0dee8017f2ee5daf9e0f03befdbd0cc4c4.